src/movies.js

```diff
@@ -1,16 +1,167 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array) {
+    const directorAll = array.map(function(movie) {
+      return movie.director;
+      })
+        return directorAll
+      }
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+    const SpielbergMovies = array.filter(function(steven){
+      if (steven.director == 'Steven Spielberg' && steven.genre.includes("Drama")) {
+          
+          return steven.director.length
+    };
+    });
+    //console.log(SpielbergMovies);
+    return SpielbergMovies.length;
+  }
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function movieRed(total, movie) {
+    if (movie.rate) {
+      return total + movie.rate;
+    } else {
+        return total;
+    }
+  }
+function ratesAverage(arr) {
+    if (arr.length === 0) {
+      return 0;
+    }
+    const totalRates = arr.reduce(movieRed, 0);
+
+    //console.log(totalRates);
+    return +(totalRates / arr.length).toFixed(2);
+  } 
+//   const avg = ratesAverage([1,2,3])
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+// You should return a number!
+// Return the rate of a single element array!
+// It should return the average of the array!
+// It should return the average of the array, float!
+// Only Drama Movies! You should return the average of Drama movies only!
+// Should return 0 if there is no Drama movie!
+function dramaMoviesRate(arr1){
+    // get only the drama movies
+    const dramaMovies = arr1.filter(function(movie){
+        if (movie.genre.includes('Drama')){
```

Comment:
Little tip to make your life easier.
Here you can simplify your code:
movie.genre.includes('Drama') represent a boolean. If the movie is a drama, movie.genre.includes('Drama') is true, if the movie is not a drama, movie.genre.includes('Drama') is false.
You could filter like this:
const dramaMovies = arr1.filter(function(movie){
        return movie.genre.includes('Drama')
    })